python/nav/web/navlets/report.py

```diff
@@ -40,15 +40,23 @@ def get_context_data(self, **kwargs):
         query_string = navlet.preferences.get('query_string')
 
         if self.mode == NAVLET_MODE_EDIT:
-            report_list = ReportList(CONFIG_FILE_PACKAGE).get_report_list()
-            report_list.extend(ReportList(CONFIG_FILE_LOCAL).get_report_list())
-            report_list.sort(key=itemgetter(1))
-            reports = {}
+            report_list = ReportList(
+                list_config_files_from_dir(CONFIG_DIR)
+            ).get_report_list()
+            report_dict = {}
             for report in report_list:
-                reports[report[0]] = (report[1],report[2])
-            report_list = []
-            for key, (val1, val2) in reports.items():
-                report_list.append((key, val1, val2))
+                report_dict[report.id] = (
+                    report.title,
+                    report.description,
+                    report.filename,
+                )
+            report_list = [
+                ReportTuple(
+                    key, report_dict[key][0], report_dict[key][1], report_dict[key][2]
+                )
+                for key in report_dict
+            ]
```

Comment:
By turning it into a dict and back into a list of ReportTuple-s we throw away all elements that have the same report.id and just keep the one last discovered

Reply:
ah, did not realize `get_report_list()` returns duplicates. 

But, this is extremely verbose code to just remove duplicates. Counter-suggestion:

```python
# remove duplicate reports by letting later versions override:
report_dict = {
    r.id: r
    for r in ReportList(list_config_files_from_dir(CONFIG_DIR)).get_report_list()
}
report_list = sorted(report_dict.values(), key=attrgetter("id"))
```

Reply:
If it doesn't matter which report wins, you could use `set()` around the list. We can't do that here since we **do** have a rule for which report wins, and for _last-found_ wins, the dict-trick or something like it is needed.

Task for the easily bored: how to remove duplicates in such a fashion that the _first-found_ wins.

Reply:
@lunkwill42 I like the first part of the suggestion, but in the second we loose our ReportTuple, which makes the following code more readable, so counter-counter suggestion:
`# remove duplicate reports by letting later versions override:
report_dict = {
    r.id: r
    for r in ReportList(list_config_files_from_dir(CONFIG_DIR)).get_report_list()
}
report_list = sorted(
                report_list=[
                    ReportTuple(key, report_dict[key][0], report_dict[key][1])
                    for key in report_dict
                ],
                key=attrgetter("id"),
            )`